src/pages/api/movies/index.ts

```diff
@@ -0,0 +1,18 @@
+import {NextApiRequest, NextApiResponse} from "next"
+import {movies} from "../../../server/models"
+import mongoose from "mongoose"
+import {sortData} from "../../../server/utils/filter"
+
+
+
+mongoose.connect(process.env.DB_URL!)
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+        const data = await movies.find();
+        const size = req.query.size as string || "9"
```

Comment:
The `size` here is a string, and not necessarily a numeric string. This means that if I pass in the string "hello", it will be happily passed along to `sortData()`, which will fail because it uses `parseInt()` without checking the return value.

A better separation of concerns would be to design `sortData()` such that it takes a `number`, and to then take responsibility here to make sure that it is a valid number, using `parseInt()` and `isNaN()`.

It's better to do it here, because if it fails this is a good place to trigger an error message before even passing it to `sortData()`, which is harder to do inside a utility function.